### ISSUE TITLE
AssignRouteAction not work

### ISSUE THREAD

**Issue by muqian3288** [esmini/esmini#275]
Dear Sir ,

    I write a xosc with AssignRouteAction. It not work in esmini.
    I wish you can help me find the reason ,If it's convenient for you.
    The following are relevant information

    esmini2.22.1
    Ubuntu18.04
    51sim (simulator) preview well.[51-sim video](https://user-images.githubusercontent.com/12215834/165242127-c4d68d4d-4c20-4298-b353-e0e3cbe097a3.mp4)
    esmini not work.[esmini-video]https://user-images.githubusercontent.com/12215834/165242193-925a694c-4c41-49ad-9c3c-01895c441038.mp4
    my xosc file[xodr_xosc.zip](https://github.com/esmini/esmini/files/8560688/xodr_xosc.zip)

    Best Wishes!

**Comment by eknabevcc** [comment]
Good point. And thanks for providing extensive information for our understanding of the problem! You're right, esmini behaves differently than 51sim in this case.

To our understanding a [Route](https://www.asam.net/static_downloads/ASAM_OpenSCENARIO_V1.2.0_Model_Documentation/modelDocumentation/content/Route.html) is used to specify the overall path for an entity throughout a road network, more specifically decide how to navigate in junctions and intersections. Without a route assigned to an entity the default controller will make random choices.

But a route does not replace other actions, actually it does not assign any control strategy (longitudinal or lateral), see [here](https://www.asam.net/index.php?eID=dumpFile&t=f&f=4908&token=ae9d9b44ab9257e817072a653b5d5e98ee0babf8#_normative_action_tables). 

So our interpretation is that a route can be assigned to make sure the vehicle makes expected routing decisions. On top of that, any action can be assigned, for example to make lane changes or speed changes, while navigating according to the route. In your case it would mean adding an explicit LaneChangeAction (or two) to the scenario.

**A bit outside the topic but good to know:**

There is a limitation in esmini default route handling. The path needs to be resolvable between waypoints without any lane changes. In other words, lane-to-lane connections needs to be available for the complete route.

However, that limitation has now been addressed by the new [FollowRoute Controller](https://esmini.github.io/#_followroute). This controller will inject lane change actions if needed and also monitor traffic to perform lane changes in a safe manner (avoiding collision with vehicles in target lane). This controller is very new, it's merged on master branch but not released yet (will be included in next release).

Although not solving your issue I hope this answer clarifies what's possible and not in esmini.
